src/main/java/sawtooth/sdk/client/signing/Secp256k1Context.java

```diff
@@ -0,0 +1,125 @@
+/* Copyright 2018 Bitwise IO
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+------------------------------------------------------------------------------*/
+
+package sawtooth.sdk.client.signing;
+
+import org.bitcoinj.core.ECKey;
+import org.bitcoinj.core.Sha256Hash;
+import org.bitcoinj.core.Utils;
+import org.spongycastle.asn1.sec.SECNamedCurves;
+import org.spongycastle.asn1.x9.X9ECParameters;
+import org.spongycastle.crypto.params.ECDomainParameters;
+import org.spongycastle.crypto.params.ECPublicKeyParameters;
+import org.spongycastle.crypto.signers.ECDSASigner;
+import org.spongycastle.math.ec.ECPoint;
+
+import java.math.BigInteger;
+import java.security.SecureRandom;
+import java.util.Arrays;
+import java.util.logging.Logger;
+
+/**
+ * A Context for signing and verifying Secp256k1 signatures.
+ */
+public class Secp256k1Context implements Context {
+  /**
+   * Logger.
+   */
+  private static final Logger LOGGER = Logger.getLogger(Context.class.getName());
+
+  /**
+   * The number of bytes in the signature.
+   */
+  private static final int NUM_SIGNATURE_BYTES = 64;
+
+  /**
+   * Half of the number of bytes in the signature.
+   */
+  private static final int HALF_NUM_SIGNATURE_BYTES = 32;
+
+  @Override
+  public final String getAlgorithmName() {
+    return "secp256k1";
```

Comment:
[nitpick] How about declaring all constants outside the scope of functions ? (similar comment for PrivateKey and PublicKey implementations below)

Reply:
Sounds good.